### PR TITLE
py3(django): defer more model imports until after Django setup (part 2)

### PR DESCRIPTION
**Pull request by joshuarli**
Continuation of https://github.com/getsentry/sentry/pull/15061. Decided to take the route of removing or weakening module inits to avoid implicit model importing before stage 2 (models.py) of the Django setup process. The imports that rely on the powerful/magical inits to hoist stuff upwards are made to be explicit.

This PR focuses on the `sentry.plugins` init and related imports, but there'll be some other things that spill out. In the next PR (part 3), `sentry.auth` init will be removed (this affects at least the apps `sentry.api.serializers.apps.Config` and `sentry.eventstream`).

**Important note:** this will definitely break sentry-plugins in some way, since import paths are changed.

### CODE REVIEW

File: src/sentry/plugins/base/bindings.py [hunk 1]
```diff
@@ -28,11 +26,15 @@ def all(self):
 
 
 class RepositoryProviderManager(ProviderManager):
-    type = providers.RepositoryProvider
+    from sentry.plugins.providers import repository
```
Comment (wedamija) [inline src/sentry/plugins/base/bindings.py]: Does this work? I'd assume that just imports straight away while loading the class definition

Comment (joshuarli) [reply src/sentry/plugins/base/bindings.py]: Why wouldn't loading the class definition block the import?

Comment (wedamija) [reply src/sentry/plugins/base/bindings.py]: I just assumed that we load the module, which will load the class, which will call this import straight away. Maybe my understanding isn't quite right though?

**Comment by joshuarli** [comment]
Mostly superseded by https://github.com/getsentry/sentry/pull/15094. Not going to delete this branch yet; keeping it around for reference.

**Pull request closed without merging**